video/out/w32_common.c

```diff
@@ -1543,6 +1582,10 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
         if (w32->opts->fullscreen != w32->current_fs)
             reinit_window_state(w32);
         return VO_TRUE;
+    case VOCTRL_GET_FULLSCREEN:
+        *(int *)arg = w32->toggle_fs ^ w32->current_fs;
+        reinit_window_state(w32);
```

Comment:
> As far as I can see VOCTRL_FULLSCREEN is only triggered on hotkey/command

Yes, that's as designed. It'll also be triggered if the user changes the fullscreen option in any other way. The only exception is if the fullscreen flag is to be updated from the VO settings, e.g.:
- user changes fullscreen "from outside" (like with win+up I guess?)
- win32 code notices this, updates its internal fullscreen state, sets VO_EVENT_FULLSCREEN_STATE
- in reaction to VO_EVENT_FULLSCREEN_STATE playloop.c sends VOCTRL_GET_FULLSCREEN
- and writes the receives value to the fullscreen option variable (without sending VOCTRL_FULLSCREEN)